src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/Models/ListViewGroupedViewModel.cs

```diff
@@ -1,571 +1,587 @@
+{
+	public class ListViewGroupedViewModel : ViewModelBase
+	{
+		Random _rnd = null;
+		private bool _areStickyHeadersEnabled;
+		private double _variableWidth;
 
-//		/// <summary>
-//		/// Group by first character
-//		/// </summary>
-//		/// <returns></returns>
-//		private IEnumerable<IGrouping<string, string>> GetGroupedSampleItems()
-//		{
-//			return _sampleItems
-//				.GroupBy(s => s[0].ToString().ToUpperInvariant())
-//				.Concat(new EmptyGroup<string, string>("This header should not appear if GroupStyle.HidesEmptyGroups is true)"));
-//		}
+		public IEnumerable<IGrouping<string, string>> GroupedSampleItems { get; }
+		public IEnumerable<object> GroupedSampleItemsAsSource { get; }
+		public IEnumerable<IGrouping<int, int>> GroupedNumericItems { get; }
+		public IEnumerable<object> GroupedNumericItemsAsSource { get; }
+		public string[] UngroupedSampleItems { get; }
+		public bool AreStickyHeadersEnabled
+		{
+			get => _areStickyHeadersEnabled;
+			set
+			{
+				_areStickyHeadersEnabled = value;
+				RaisePropertyChanged();
+			}
+		}
 
-//		internal static async Task<IEnumerable<object>> GetAsCollectionViewSource<T1, T2>(CancellationToken ct, IEnumerable<IGrouping<T1, T2>> groups)
-//		{
-//			ICollectionView source = null;
+		public IEnumerable<object> GroupedTownsAsSource { get; }
+		public IEnumerable<object> EmptyGroupsAsSource { get; }
+		public IEnumerable<object> EmptyAsSource { get; }
 
-//#if XAMARIN
-//			var dispatcher = CoreDispatcher.Main;
-//#else
-//			var dispatcher = CoreApplication.MainView.Dispatcher;
-//#endif
-//			// Have to create CollectionViewSource on UI thread
-//			await dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
-//			{
-//				source = new CollectionViewSource()
-//				{
-//					Source = groups,
-//					IsSourceGrouped = true
-//				}.View;
-//			}).AsTask(ct);
-//			return source;
-//		}
+		public double VariableWidth
+		{
+			get => _variableWidth;
+			set
+			{
+				_variableWidth = value;
+				RaisePropertyChanged();
+			}
+		}
 
-//		private IEnumerable<IGrouping<int, int>> GetGroupedNumericItems()
-//		{
-//			return _sampleNumbers
-//				.GroupBy(i => GetFirstDigit(i))
-//				.OrderBy(g => g.Key)
-//				.Concat(new EmptyGroup<int, int>(1111111111)); //Should not appear if GroupStyle.HidesEmptyGroups is true
-//		}
+		public ListViewGroupedViewModel(CoreDispatcher dispatcher) : base(dispatcher)
+		{
+			_rnd = new Random();
 
-//		private int GetFirstDigit(int number)
-//		{
-//			number = Math.Abs(number);
-//			while (number >= 10)
-//			{
-//				number /= 10;
-//			}
-//			return number;
-//		}
+			GroupedSampleItems = GetGroupedSampleItems();
+			GroupedSampleItemsAsSource = ListViewViewModel.GetAsCollectionViewSource(GetGroupedSampleItems());
+			GroupedNumericItems = GetGroupedNumericItems();
+			GroupedNumericItemsAsSource = ListViewViewModel.GetAsCollectionViewSource(GetGroupedNumericItems());
+			//TODO: Restore dynamic samples based on IObservable
```

Comment:
If the issue was just keeping the `[]` syntax in view, you can inject with a `Dictionary<string, object>` and expose that from the vm via an indexer:
```
public object this [string key] => _dynamicProperties.TryGetValue(key, out var value) ? value : default;
```
As for rx, I'm not sure how pure we want to keep the samples..?

Reply:
Backwards support for the indexer syntax is already there. :) https://github.com/nventive/Uno/blob/a653d475b6010858b3d7a1878dd5070e863d0eae/src/SamplesApp/UITests.Shared/Helpers/ViewModelBase.cs#L68